### PR TITLE
Refresh without sendable trait

### PR DESCRIPTION
**Pull request by inetic**
Simple refresh mechanism without traits

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/368)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_membrane.rs [hunk 1]
```diff
@@ -553,6 +558,23 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
     }
 
+    fn handle_refresh(&mut self, header: MessageHeader, body: Bytes) -> RoutingResult {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: Can this be handler section

File: src/messages/refresh.rs [hunk 2]
```diff
@@ -15,28 +15,26 @@
...
 // relating to use of the SAFE Network Software.
 
-#![allow(unused_assignments)]
-
 use cbor::CborTagEncode;
 use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
-use sendable::Sendable;
 
 #[derive(PartialEq, Eq, Clone, Debug)]
-pub struct Refresh<T> where T: Sendable + Encodable + Decodable {
```
Comment (benjaminbollen) [inline src/messages/refresh.rs]: Why are we dropping the template?

File: src/routing_membrane.rs (hunk 1 above)
Comment (inetic) [inline src/routing_membrane.rs]: Done

File: src/node_interface.rs [hunk 3]
```diff
@@ -34,6 +34,7 @@ pub enum MethodCall {
     Post,
     /// request to refresh
     Refresh { content: Box<Sendable>, },
+    RefreshNew { type_tag: u64, from_group: NameType, payload: Vec<u8>, },
```
Comment (benjaminbollen) [inline src/node_interface.rs]: Go ahead and already rename the MethodCall::Refresh with the new signature; it ensures that we have a unique behaviour for this Refresh call

File: src/routing_membrane.rs [hunk 4]
```diff
@@ -890,6 +911,7 @@ impl<F> RoutingMembrane<F> where F: Interface {
             MethodCall::Put { destination: x, content: y, } => self.put(x, y),
             MethodCall::Get { type_id: x, name: y, } => self.get(x, y),
             MethodCall::Refresh { content: x, } => self.refresh(x),
+            MethodCall::RefreshNew { type_tag, from_group, payload } => self.refresh_new(type_tag, from_group, payload),
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: Same here, just replace the Refresh call

File: src/routing_membrane.rs [hunk 5]
```diff
@@ -964,6 +986,31 @@ impl<F> RoutingMembrane<F> where F: Interface {
         Ok(())
     }
 
+    fn handle_refresh(&mut self, header: MessageHeader, body: Bytes) -> RoutingResult {
+        let refresh = try!(decode::<Refresh>(&body));
+        let from_group = try!(header.from_group().ok_or(RoutingError::RefreshNotFromGroup));
+
+        if from_group != refresh.from_group {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: also call `our_authority` and chech it returns our_authority = OurCloseGroup

**Comment by benjaminbollen** [comment]
looks good to me

**Comment by benjaminbollen** [comment]
resolves #363

**Pull request merged**